### PR TITLE
Pinmux device name changes

### PR DESCRIPTION
**Pull request by bigdinotech**
-change pinmux name to be compatible with Zephyr 1.7

### CODE REVIEW

**Comment by bigdinotech** [comment]
@eriknyquist @SidLeung please review

**Review by eriknyquist — APPROVED** [review]

File: common/arduino/variant.c [hunk 1]
```diff
@@ -24,6 +24,8 @@
 #define PRINT           printk
 #endif
 
+#define PINMUX_NAME CONFIG_PINMUX_NAME
```
Comment (eriknyquist) [inline common/arduino/variant.c:27]: Actually, I take my approval back.... what purpose is this define serving?

We already have "CONFIG_PINMUX_NAME", which everyone will understand instantly (it's a config file option).

Re-defining it without the CONFIG_ part is just confusing.

**Review by eriknyquist — CHANGES_REQUESTED** [review]
Don't see the point of re-defining CONFIG_PINMUX_NAME to a slightly different name

File: common/arduino/variant.c (hunk 1 above)
Comment (bigdinotech) [reply common/arduino/variant.c:27]: Just copying what Zephyr did in their example.

Comment (eriknyquist) [reply common/arduino/variant.c:27]: That doesn't mean it's required, or good. My opinion is that we should remove it.  We're not gaining any abstraction or de-cluttering any of the code by using that define. It's just an extra, unnecessary layer of indirection that requires extra effort from the reader. I feel the same about "printk-->PRINT" but that's an unrelated matter. So I won't approve that, unless you have a good reason for it that you can share?

**Comment by eriknyquist** [comment]
Alright I'm must merging this, we need it

**Pull request merged**